File: docs/swarm/turing-model/old/ModelSwarm.java
import swarm.*;
import swarm.activity.*;
import swarm.defobj.*;
import swarm.objectbase.*;
import swarm.space.*;
import swarm.collections.*;

public class ModelSwarm extends SwarmImpl{
	public int worldXSize, worldYSize;
	
	ActivatorSpace activator;
	TemporarySpace inhibitor;
	TemporarySpace inhibitorTmp;
	TemporarySpace activatorTmp;
	
	ActionGroup modelActions;
	Schedule modelSchedule;
	
	public ModelSwarm(Zone aZone){
		super(aZone);
		EmptyProbeMap probeMap;
		worldXSize=80;
		worldYSize=80;
		
		probeMap=new EmptyProbeMapImpl(aZone,this.getClass());
		probeMap.addProbe(Globals.env.probeLibrary.getProbeForVariable$inClass
          ("worldXSize",getClass()));
		probeMap.addProbe(Globals.env.probeLibrary.getProbeForVariable$inClass
          ("worldYSize",getClass()));
		
        Globals.env.probeLibrary.setProbeMap$For(probeMap,this.getClass());
	}
	
	public Object buildObjects(){
		inhibitor=new TemporarySpace(this,worldXSize,2);
		inhibitor.initializeLattice();
		
		inhibitorTmp=new TemporarySpace(this,worldXSize,2);
		inhibitorTmp.initializeLattice();
		
		activatorTmp=new TemporarySpace(this,worldXSize,2);
		activatorTmp.initializeLattice();
		
		activator=new ActivatorSpace(getZone(),worldXSize,worldYSize);
		//activator.setNumStates(2);
		activator.setInhibitor(inhibitor);
		activator.setInhibitorTmp(inhibitorTmp);
		activator.setActivatorTmp(activatorTmp);
		activator.setRule(0.001,2.,11.,0.,0.3,3,8,0.,1.,0.);
		
		return this;
	}
	
	public Object buildActions(){
		modelActions=new ActionGroupImpl(this);
		try{
			modelActions.createActionTo$message(activator,
				new Selector(activator.getClass(),"stepRule",false));
		} catch (Exception e) {
			e.printStackTrace (System.err);
			System.exit(1);
		}
		
		modelSchedule=new ScheduleImpl(this,1);
		modelSchedule.at$createAction(0,modelActions);
		return this;
	}
	
	public Activity activateIn(Swarm context){
    	super.activateIn (context);
    	modelSchedule.activateIn(this);
		return getActivity();
	}
	
	public ActivatorSpace getWorld(){
		return activator;
	}
}

File: docs/swarm/turing-model/old/ModelSwarm.m
#import "ModelSwarm.h"
#import <random.h>
#import <activity.h>
#import <collections.h>

@implementation ModelSwarm

- getWorld
{
	return activator;
}


+ createBegin: aZone 
{
	ModelSwarm *obj;
	id <ProbeMap> probeMap;
	
	obj = [super createBegin: aZone];
	obj->worldXSize = 80;
	obj->worldYSize = 80;
	
	probeMap = [EmptyProbeMap createBegin: aZone];
	[probeMap setProbedClass: [self class]];
	probeMap = [probeMap createEnd];
	
	[probeMap addProbe: [probeLibrary getProbeForVariable: "worldXSize"
	                                    inClass: [self class]]];
	[probeMap addProbe: [probeLibrary getProbeForVariable: "worldYSize"
	                                    inClass: [self class]]];
	
	[probeLibrary setProbeMap: probeMap For: [self class]];
	
	return obj;
}


- createEnd
{
	return [super createEnd];
}


- buildObjects
{
	inhibitor=[TemporarySpace createBegin: self];
	[inhibitor setSizeX: worldXSize Y: 2];
	inhibitor=[inhibitor createEnd];
	[inhibitor initializeLattice];
	
	inhibitorTmp=[TemporarySpace createBegin: self];
	[inhibitorTmp setSizeX: worldXSize Y: 2];
	inhibitorTmp=[inhibitorTmp createEnd];
	[inhibitorTmp initializeLattice];
	
	activatorTmp=[TemporarySpace createBegin: self];
	[activatorTmp setSizeX: worldXSize Y: 2];
	activatorTmp=[activatorTmp createEnd];
	[activatorTmp initializeLattice];
	
	activator = [ActivatorSpace createBegin: self];
	[activator setSizeX: worldXSize Y: worldYSize];
	[activator setNumStates: 2];
	[activator setInhibitor: inhibitor];
	[activator setInhibitorTmp: inhibitorTmp];
	[activator setActivatorTmp: activatorTmp];
	activator = [activator createEnd];
	
	[activator setRule_p: 0.002 w1: 8.0 w2: 21.0 m0: 0.0 m1: 1.0 ru: 1 rv: 0 d: 0 e: 1.0 initialProb: 0.01];
	
	return self;
}


- buildActions
{
	modelActions = [ActionGroup create: self];
	[modelActions createActionTo: activator    message: M(stepRule)];
	
	modelSchedule = [Schedule createBegin: self];
	[modelSchedule setRepeatInterval: 1];
	modelSchedule = [modelSchedule createEnd];
	[modelSchedule at: 0 createAction: modelActions]; 
	
	return self;
}


- activateIn: swarmContext
{
	[super activateIn: swarmContext];
	[modelSchedule activateIn: self];
	
	return [self getSwarmActivity];
}
@end
